src/geo/transform.js

```diff
@@ -385,6 +385,47 @@ class Transform {
             this.zoom)._zoomTo(zoom);
     }
 
+    /**
+     * Get the latitude and longitude under the camera, the altitude of the camera in meters, bearing & pitch
+     * @returns {Object} containing longitude, latitude, altitude and pitch(in degrees)
+     */
+    getCameraPosition(): { lng: number, lat: number, altitude: number, pitch: number, bearing: number } {
+        const pitch = this._pitch;
+        const altitude = Math.cos(pitch) * this.cameraToCenterDistance;
+        const latOffset = Math.tan(pitch) * this.cameraToCenterDistance;
+        const latPosPointInPixels = this.centerPoint.add(new Point(0, latOffset));
+        const latLong = this.pointLocation(latPosPointInPixels);
+        const verticalScaleConstant = this.worldSize / (2 * Math.PI * 6378137 * Math.abs(Math.cos(latLong.lat * (Math.PI / 180))));
```

Comment:
Can we add a comment here explaining the magic number `6378137` and what this constant represents?

Reply:
@ansis it's been ages since we whipped this up; do you recall what was up with this magic # ?

Reply:
I've expanded this and added a comment. Per https://github.com/mapbox/mapbox-gl-js/issues/8777#issuecomment-533246706 we could eventually move this to a global constant, but right now I'm just cleaning up this stale PR to investigate if it could still be merged or not.